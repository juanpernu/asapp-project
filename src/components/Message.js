/** React dependencies */
import React, { useEffect, useRef } from 'react';

/** Component dependencies */
import Icon from './Icons';

/**
 * Stateless Component that draw a box with the message.
 * @param {String} message
 * @param {Object} from
 * @param {String} timestamp
 * @param {Number} user_id
 * @param {Boolean} saw_message
 */
const Message = ({ message, from, timestamp, user_id, saw_message }) => {
  const selfMessageClass = from.id === user_id && 'self-message';
  const sawMessage = saw_message ? 'blue' : 'black';

  const thisMessage = useRef();
  useEffect(() => {
    thisMessage.current.scrollIntoView();
  }, [message]);

  return (
    <div className={`message ${selfMessageClass}`} ref={thisMessage}>
      <span className="message-text">{message}</span>
      <div className="message-data">
        <span className="message-data--timestamp">{timestamp}</span>
        {selfMessageClass && <Icon type='check' colorFill={sawMessage} />}
      </div>
    </div>
  )
}

export default Message;
