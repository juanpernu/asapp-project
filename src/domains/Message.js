/**
 * Class for Messages. Represents and handles all the
 * attributes for the data of the messages.
 */
class Message {
  /**
   * The constructor of the class.
   * @constructor
   *
   * @param {Object} text
   * @param {Object} from
   * @param {Object} timestamp
   * @param {Object} user_id
   * @param {Object} saw_message
   */
  constructor(text, from, timestamp, user_id, saw_message) {
    this.text = text;
    this.from = from;
    this.timestamp = timestamp;
    this.user_id = user_id;
    this.saw_message = saw_message;
  }
}

export default Message;
