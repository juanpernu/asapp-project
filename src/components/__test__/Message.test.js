import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Message from '../Message';

const mockedData = {
  message: 'Test message',
  from: {
    id: 1,
  },
  timestamp: '11:11',
  user_id: 1,
  saw_message: false
}

describe('Componentes - Message', () => {
  it('Should render correctly', () => {
    window.HTMLElement.prototype.scrollIntoView = function () {};
    const { getByText } = render(
    <Message
      message={mockedData.message}
      from={mockedData.from}
      timestamp={mockedData.timestamp}
      user_id={mockedData.user_id}
      saw_message={mockedData.saw_message}/>);
    expect(getByText('Test message')).toBeInTheDocument();
    expect(getByText('11:11')).toBeInTheDocument();
    expect(Message).toMatchSnapshot();
  });
});
