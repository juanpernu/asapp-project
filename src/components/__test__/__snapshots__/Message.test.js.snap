// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componentes - Message Should render correctly 1`] = `[Function]`;
